File: src/main/java/path_generator/discrete_function/DiscreteFunctionPeople.java
package path_generator.discrete_function;

import java.util.ArrayList;
import java.util.List;

public class DiscreteFunctionPeople implements DiscreteFunction {
    private final List<Double> function;
    public static final Integer NUMBER_OF_SECTIONS = 24;
    public static final Double FUNCTION_MINIMUM_VALUE = 0.0;
    public static final Double FUNCTION_MAXIMUM_VALUE = 1.0;
    /**
     * Задает любую функцию с 24 отрезками.
     * @param function
     */
    public DiscreteFunctionPeople(List<Double> function) {
        if (function.size() != NUMBER_OF_SECTIONS) {
            throw new IllegalArgumentException("List have size isn't 24.");
        }
        for (int i = 0; i < NUMBER_OF_SECTIONS; i++) {
            Double currentValue = function.get(i);
            if (currentValue < FUNCTION_MINIMUM_VALUE) {
                throw new IllegalArgumentException("Function value is less than the minimum value.");
            }
            if (currentValue > FUNCTION_MAXIMUM_VALUE) {
                throw new IllegalArgumentException("Function value is greater than the maximum value.");
            }
        }
        this.function = function;
    }

    /**
     * Задает функцию с константным значением.
     * @param constanta
     */
    public DiscreteFunctionPeople(final Double constanta) {
        if (constanta < FUNCTION_MINIMUM_VALUE) {
            throw new IllegalArgumentException("Function value is less than the minimum value.");
        }
        if (constanta > FUNCTION_MAXIMUM_VALUE) {
            throw new IllegalArgumentException("Function value is greater than the maximum value.");
        }
        this.function = new ArrayList<>();
        for (int i = 0; i < NUMBER_OF_SECTIONS; i++) {
            this.function.add(constanta);
        }
    }


    @Override
    public Double getIthPriority(Integer ithHour) {
        return this.function.get(ithHour);
    }
}

File: simulator/src/main/kotlin/route_generator/simple_model_generator/discrete_function/DiscreteFunctionPeople.kt
package route_generator.simple_model_generator.discrete_function

import jakarta.validation.constraints.NotNull

class DiscreteFunctionPeople : DiscreteFunction<Double?> {
    private val function: MutableList<Double>

    /**
     * Задает любую функцию с 24 отрезками.
     * @param function
     */
    constructor(function: @NotNull MutableList<Double>) {
        require(function.size == DiscreteFunction.NUMBER_OF_SECTIONS) { "List have size isn't 24." }
        for (i in 0 until DiscreteFunction.NUMBER_OF_SECTIONS) {
            val currentValue = function[i]
            require(!(currentValue < FUNCTION_MINIMUM_VALUE)) { "Function value is less than the minimum value." }
            require(!(currentValue > FUNCTION_MAXIMUM_VALUE)) { "Function value is greater than the maximum value." }
        }
        this.function = function
    }

    /**
     * Задает функцию с константным значением.
     * @param constanta
     */
    constructor(constanta: Double) {
        require(!(constanta < FUNCTION_MINIMUM_VALUE)) { "Function value is less than the minimum value." }
        require(!(constanta > FUNCTION_MAXIMUM_VALUE)) { "Function value is greater than the maximum value." }
        this.function = ArrayList()
        for (i in 0 until DiscreteFunction.NUMBER_OF_SECTIONS) {
            function.add(constanta)
        }
    }


    override fun getIthPriority(ithHour: Int): Double {
        return function[ithHour]
    }

    companion object {
        const val FUNCTION_MINIMUM_VALUE: Double = 0.0
        const val FUNCTION_MAXIMUM_VALUE: Double = 1.0
    }
}
